staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -129,7 +129,7 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 	clientCertificateExpirationHistogram.Observe(remaining.Seconds())
 	chains, err := req.TLS.PeerCertificates[0].Verify(optsCopy)
 	if err != nil {
-		return nil, false, err
+		return nil, false, fmt.Errorf("verifying certificate %v failed: %w", req.TLS.PeerCertificates[0].SerialNumber, err)
```

Comment:
@mikedanese thoughts on including the Serial+CN of the cert failing validation here?